cleverhans/attacks_eager.py

```diff
@@ -0,0 +1,331 @@
+from abc import ABCMeta
+import numpy as np
+from six.moves import xrange
+import warnings
+import collections
+import tensorflow as tf
+from distutils.version import LooseVersion
+import cleverhans.utils as utils
+
+_logger = utils.create_logger("cleverhans.attacks_eager")
+
+if LooseVersion(tf.__version__) < LooseVersion('1.8.0'):
+    error = ('For eager execution',
+             'use Tensorflow version greather than 1.8.0.')
+    print(error)
+    exit(1)
+
+try:
+    from cleverhans.model_eager import Model_Eager
+except ImportError:
+    error = 'Import Error: cleverhans.model_eager'
+    raise ImportError(error)
+
+
+class Attack(object):
+    """
+    Abstract base class for all attack classes.
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, model, back='tfe', dtypestr='float32'):
+        """
+        :param model: An instance of the cleverhans.model.Model_Eager class.
+        :param back: The backend to use. Currently 'tfe' is the only option.
+        :param dtypestr: datatype of the input data samples and crafted
+                        adversarial attacks.
+        """
+        # Validate the input arguments.
+        if not(back == 'tfe'):
+            raise ValueError("Backend argument must be 'tfe'.")
+        if back == 'tfe':
```

Comment:
Isn't this if statement always true?

Reply:
True, redundant snippet is used to stay similar to previous code in attacks.py 
`        if not(back == 'tf'):
            raise ValueError("Backend argument must be 'tf'.")

        if back == 'tf':
            import tensorflow as tf
            self.tf_dtype = tf.as_dtype(dtypestr)`
strictly speaking this is not necessary, let me know what you think.

Reply:
I don't know I didn't review the previous code but it seems redundant just the same. No need to clean all existing code but writing new code you can fix stuff that you notice along the way.

Reply:
Sure, will make the changes in updated PR.